dot/core/service.go

```diff
@@ -282,11 +280,14 @@ func (s *Service) handleBlocksAsync() {
 				continue
 			}
 
+			prev := s.blockState.BestBlockHash()
 			if err := s.handleChainReorg(prev, block.Header.Hash()); err != nil {
 				logger.Warnf("failed to re-add transactions to chain upon re-org: %s", err)
 			}
 
-			s.maintainTransactionPool(block)
+			if err := s.maintainTransactionPool(block); err != nil {
+				logger.Warnf("failed to maintain txn pool after re-org: %s", err)
```

Comment:
I wonder if we ever get to this case.  Maybe it would be best to panic until we figure out reporting errors from `handleBlocksAsync`.

Reply:
Will leave panics and am updating maintaintxnPool to return more errors when appropriate